src/dog/services/dog/dog.service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from '@nestjs/common';
+import { DOGS } from 'src/dog/datasource/dogs';
+
+@Injectable()
+export class DogService {
+  find(dogId: number) {
+    let d: any;
+    DOGS.forEach((dog) => {
+      if (dog.id == dogId) {
+        d = dog;
+      }
+    });
+    return d;
+  }
+  findOne(age: number, race: string) {
+    let d: any;
+    DOGS.forEach((dog) => {
+      if (dog.age == age && dog.race == race) {
+        d = dog;
+      }
+    });
+    return d;
+  }
+  findAll() {
+    return DOGS;
+  }
+  create(age: number, race: string, color: string) {
+    const id = DOGS.reduce((maxId, dog) => Math.max(maxId, dog.id), 0) + 1;
+    const dogAdd = { id: id, age: age, race: race, color: color };
+    DOGS.push(dogAdd);
+    return dogAdd;
+  }
+  update(dogId: number, age: number) {
+    let d: any;
+    DOGS.forEach((dog) => {
+      if (dog.id == dogId) {
+        dog.age = age;
+        d = dog;
+      }
+    });
+    return d;
```

Comment:
Same comment as in the other search functions. Please improve the performance of this iteration.